src/bid/score.rs

```diff
@@ -5,22 +5,107 @@
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
 //! Score generation
-#![cfg(feature = "std")]
 
-use super::Score;
+#[cfg(feature = "canon")]
+use canonical::Canon;
+#[cfg(feature = "canon")]
+use canonical_derive::Canon;
+
+#[cfg(feature = "std")]
 use crate::bid::Bid;
+#[cfg(feature = "std")]
 use crate::errors::BlindBidError;
+use core::ops::Deref;
 use dusk_bls12_381::BlsScalar;
-use dusk_bytes::Serializable;
+use dusk_bytes::{DeserializableSlice, Serializable};
+#[cfg(feature = "std")]
 use dusk_jubjub::JubJubAffine;
+#[cfg(feature = "std")]
 use dusk_plonk::prelude::*;
+#[cfg(feature = "std")]
 use num_bigint::BigUint;
+#[cfg(feature = "std")]
 use num_traits::{One, Zero};
+#[cfg(feature = "std")]
```

Comment:
It's okay-ish like that, but it might be better put all the std usage under one single module (to have only one `#[cfg(feature = "std")` or maybe use `cfg-if` macro (https://docs.rs/cfg-if/1.0.0/cfg_if/).

It's a bit strange we have in one module both std and no_std in this way, but if it works we can rethink about it later.

Reply:
It's super ugly. I wanted to comment that also in the meeting but we were already quite long on it. 

cfg-if sounds like a good option

Reply:
`cfg-if` is not perfect as well, but for this massive usage it's good enough.